ClavaLaraApi/src-lara-clava/clava/clava/pass/SingleReturnFunction.js

```diff
@@ -0,0 +1,41 @@
+laraImport("lara.pass.Pass");
+laraImport("weaver.Query");
+
+class SingleReturnFunction extends Pass {
+  constructor() {
+    super("SingleReturnFunction");
+  }
+
+  _apply_impl($jp) {
+    if ($jp.joinpointType !== "function" || !$jp.isImplementation) {
+      return;
+    }
+    const $body = $jp.body;
+    const $returnStmts = Array.from(Query.searchFrom($body, "returnStmt"));
+    if ($returnStmts.length <= 1) {
+      return;
+    }
+
+    $body.insertEnd("__returnLabel:");
+
+    const returnType = $jp.returnType;
+    const returnIsVoid =
```

Comment:
The name of the variable is not reversed? According to the test, the result seems to be "returnIsNotVoid". But I would recomend to maintain this name, and change the tests, for instance: 

returnType.isBuiltin && returnType.builtinKind === "Void"